### ISSUE TITLE
install: hierarchical src results in hierarchical run

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#4643]
> Low priority bug with minimal impact.

The original plan was that the way a user organises their flows in the src dir should not impact the way they were installed into the run dir.

Somewhere along the line there was a regression:

```
~/cylc-src/
`-- a/
    `-- b/
```

```console
$ cylc install a/b
INSTALLED a/b/run1 from a/b
```

This is actually kinda a nice thing in some ways, and not in others, depends on working practices. Perhaps we should consider how to support both.

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by wxtim** [comment]
I think we should:
- Close this issue
- Open an issue describing replacement of --directory/-C and do it ASAP.
- Open an issue describing the idea of a shortcut to --flow-name=basename to implement if we find a user demand for it: Feature won't break any existing functionality so it's low priority.
- Open a docs issue to check for references to cylc install a/b/c installing to ~/cylc-run/c.

**Comment by wxtim** [comment]
Closed with sucessors
